### ISSUE TITLE
Nativescript 4.0 | Issue with new App_resources / nsconfig

### ISSUE THREAD

**Issue by breningham** [EddyVerbruggen/nativescript-plugin-firebase#677]
I have just created a new Nativescript 4 Project, and i am unable to get it to work with this plugin.

The issue is that with the new App_Resources layout, it seems that the google-services.json is not being copied over and we fail with a 'google-services.json' not found.

was just wondering if anyone has any solution or workaround to solve this?

**Comment by breningham** [comment]
Just to add some more context.

the Issue is caused after running `tns resources update android` to attempt to fix a similar issue with Google Maps. Firebase worked before the code was ran.

**Comment by breningham** [comment]
I seem to have solved the issue by replacing the `firebase-copy-google-services.js` contents with this:
```(javascript)

var path = require("path");
var fs = require("fs");

module.exports = function() {

    var nsconfigpath = path.join(__dirname, "..", "..", "nsconfig.json");
    var sourceGoogleJson = path.join(__dirname, "..", "..", "src", "resources", "Android", "google-services.json");

    if ( fs.existsSync(nsconfig) ) {
        console.log('nsconfig exists! using configured appResourcesPath');
        var nsconfig = require(nsconfigpath);
        if ( nsconfig.appResourcesPath ) {
            sourceGoogleJson = path.join(__dirname, "..", "..", nsconfig.appResourcesPath, "Android", "google-services.json");
        }
    } 

    var destinationGoogleJson = path.join(__dirname, "..", "..", "platforms", "android", "google-services.json");
    if (fs.existsSync(sourceGoogleJson) && fs.existsSync(path.dirname(destinationGoogleJson))) {
        console.log("Copy " + sourceGoogleJson + " to " + destinationGoogleJson + ".");
        fs.writeFileSync(destinationGoogleJson, fs.readFileSync(sourceGoogleJson));
    }
};
```

**Comment by EddyVerbruggen** [comment]
Ah, thanks for the heads-up! So you have something like this, right?

<img width="649" alt="screen shot 2018-04-25 at 16 50 02" src="https://user-images.githubusercontent.com/1426370/39253699-b9e4e28a-48a8-11e8-9ba8-b528d4e3e355.png">

**Comment by EddyVerbruggen** [comment]
Hmm, I actually wonder why it didn't work for you because the hook uses `$projectData.appResourcesDirectoryPath` to determine the resources path and in my demo project it point to the correct folder (the one [my `nsconfig.json`](https://github.com/EddyVerbruggen/nativescript-plugin-firebase/blob/master/demo/nsconfig.json) points to). So `path.join($projectData.appResourcesDirectoryPath, "Android", "google-services.json")` resolves to the correct location for me. Thoughts?

**Comment by breningham** [comment]
The contents of my nsconfig.json is 
```javascript
{
    "appPath": "src/app",
    "appResourcesPath": "src/resources"
}
```
might have something to do with the path? + sub path? 

Though to be honest, by default the `hooks/after-prepare/firebase-copy-google-services.js`

looks like this : 
```javascript

var path = require("path");
var fs = require("fs");

module.exports = function() {

    var sourceGoogleJson = path.join(__dirname, "..", "..", "app", "App_Resources", "Android", "google-services.json");
    var destinationGoogleJson = path.join(__dirname, "..", "..", "platforms", "android", "google-services.json");
    if (fs.existsSync(sourceGoogleJson) && fs.existsSync(path.dirname(destinationGoogleJson))) {
        console.log("Copy " + sourceGoogleJson + " to " + destinationGoogleJson + ".");
        fs.writeFileSync(destinationGoogleJson, fs.readFileSync(sourceGoogleJson));
    }
};

```

So maybe im installing the wrong version?

**Comment by breningham** [comment]
after installing 5.3.0 i get the following error: 
node_modules/nativescript-plugin-firebase/firebase.d.ts(699,1): error TS1036: Statements are not allowed in ambient contexts.

Seems to be `exports.analytics = analytics;`

**Comment by EddyVerbruggen** [comment]
@breningham The analytics export is fixed in 5.3.1.
